subsys/bluetooth/host/classic/sdp.c

```diff
@@ -978,12 +978,13 @@ static uint16_t create_attr_list(struct bt_sdp *sdp, struct bt_sdp_record *recor
  *   IDs, the lower 2 bytes contain the ID and the upper 2 bytes are set to
  *   0xFFFF. For attribute ranges, the lower 2bytes indicate the start ID and
  *   the upper 2bytes indicate the end ID
+ *  @param filter_count Max element slots of filter to be filled in
  *  @param num_filters No. of filter elements filled in (to be returned)
  *
  *  @return 0 for success, or relevant error code
  */
 static uint16_t get_att_search_list(struct net_buf *buf, uint32_t *filter,
-				 uint8_t *num_filters)
+				 uint8_t filter_count, uint8_t *num_filters)
```

Comment:
`size_t filter_count` would be the more correct type, rather than uint8_t. It's also the type that `ARRAY_SIZE()` returns. I'd also rename this since `num_filters` and `filter_count` sound too similar. I'd propose to rename `filter_count` to `max_filters` instead.

Reply:
I think only change `uint8_t filter_count` to `size_t filter_count` is not reasonable. Because, the type of `num_filters` is `uint8_t`.
There are risks in comparing two different types of value. Such as, `*num_filters` always less than `filter_count` if the filter_count is more than 0xFF.

So if `uint8_t filter_count` is changed to `size_t filter_count`, `uint8_t *num_filters` also need to be changed to `size_t *num_filters`.

Reply:
Chang `filter_count` to `max_filters`

Reply:
> `size_t filter_count` would be the more correct type, rather than uint8_t. It's also the type that `ARRAY_SIZE()` returns. I'd also rename this since `num_filters` and `filter_count` sound too similar. I'd propose to rename `filter_count` to `max_filters` instead.

What exactly is the problem or "chain reaction" you're referring to? You're right that both should use `size_t` but what's the problem with that? The stuff that you now introduced with the ternary operator makes the code less readable IMO.

Reply:
`num_filters` will be passed to `sad.num_filters` in function `create_attr_list`. The changes will occur in the functions `create_attr_list` and `select_attrs`.

Reply:
> `sad.num_filters`

But then you'd have just one place to do a typecast to `uint8_t`, i.e. when calling `create_attr_list()`, right? That sounds simpler to me than what you now did. Remember that the code is hardcoding the array size to *10* so there's no reason to unnecessarily complicate the code with additional bounds checks. You could throw in there a build assert of course, if you want.

Reply:
Well. I have made a change to make sure the type of all of variables related to array `filter` is `size_t`.